aws-amplify/amplify-category-api
# chore: address pentest findings & add non GSI e2e tests

#### Description of changes

- Addressed pentest findings
  - Index name format check to avoid injection
  - Remove response URL in log
  - Add allow list when converting string type to boolean or number
- Added construct e2e test for multiple GSI updates + other non GSI updates. The tests are passed locally.



##### CDK / CloudFormation Parameters Changed



#### Issue #, if available



#### Description of how you validated changes

#### Checklist



- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
